src/web-auth/redirect.js

```diff
@@ -1,74 +1,33 @@
 var Warn = require('../helper/warn');
-var assert = require('../helper/assert');
 
-function Redirect(client, options) {
+function Redirect(auth0, options) {
   this.baseOptions = options;
-  this.client = client;
+  this.client = auth0.client;
+  this.crossOriginAuthentication = new CrossOriginAuthentication(auth0, this.baseOptions);
 
   this.warn = new Warn({
     disableWarnings: !!options._disableDeprecationWarnings
   });
 }
 
 /**
- * @callback credentialsCallback
- * @param {Error} [err] error returned by Auth0 with the reason of the Auth failure
- * @param {Object} [result] result of the AuthN request
- * @param {String} result.accessToken token that can be used with {@link userinfo}
- * @param {String} [result.idToken] token that identifies the user
- * @param {String} [result.refreshToken] token that can be used to get new access tokens from Auth0. Note that not all clients can request them or the resource server might not allow them.
- */
-
-/**
- * Performs authentication with username/email and password with a database connection
- *
- * This method is not compatible with API Auth so if you need to fetch API tokens with audience
- * you should use {@link authorize} or {@link login}.
+ * Logs in the user with username and password using the cross origin authentication (/co/authenticate) flow. You can use either `username` or `email` to identify the user, but `username` will take precedence over `email`.
+ * This only works when 3rd party cookies are enabled in the browser. After the /co/authenticate call, you'll have to use the {@link parseHash} function at the `redirectUri` specified in the constructor.
  *
  * @method loginWithCredentials
- * @param {Object} options
- * @param {String} [options.redirectUri] url that the Auth0 will redirect after Auth with the Authorization Response
- * @param {String} [options.responseType] type of the response used. It can be any of the values `code` and `token`
- * @param {String} [options.responseMode] how the AuthN response is encoded and redirected back to the client. Supported values are `query` and `fragment`
- * @param {String} [options.scope] scopes to be requested during AuthN. e.g. `openid email`
- * @param {credentialsCallback} cb
+ * @deprecated This method will be released in the next major version. Use `webAuth.login` instead.
+ * @param {Object} options options used in the {@link authorize} call after the login_ticket is acquired
+ * @param {String} [options.username] Username (mutually exclusive with email)
+ * @param {String} [options.email] Email (mutually exclusive with username)
+ * @param {String} options.password Password
+ * @param {String} [options.connection] Connection used to authenticate the user, it can be a realm name or a database connection name
+ * @param {crossOriginLoginCallback} cb Callback function called only when an authentication error, like invalid username or password, occurs. For other types of errors, there will be a redirect to the `redirectUri`.
  */
```

Comment:
What about responseType, scope, etc?

Shouldn't we remove the deprecated flag?

Reply:
Moving forward, we'll end up removing .popup and .redirect, right? So, ideally, let's warn about this being deprecated and remove it in the next major. What do you think?

About other options, we explain that all options that can be sent to `authorize`, can be sent here: 
`@param {Object} options options used in the {@link authorize} call after the login_ticket is acquired`